### ISSUE TITLE
Allow support for trailing commas in JSON blocks (currently those break the JSON ZED config readout)

### ISSUE THREAD

**Issue by krstp** [zed-industries/community#1322]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Trailing commas DO break JSON. This appears a known issue to ZED team as per reported solution in #1312. As a solution it is understood, the trailing single comma should be probably allowed, especially in Python, this is kind of critical for code formatting when using tools such as `black`.

### Environment

Zed: v0.79.1 (stable)
OS: macOS 13.2.1
Memory: 32 GiB
Architecture: x86_64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Sample code that shows last JSON attribute trailing comma, such as:
```
{
[...]
    "remove_trailing_whitespace_on_save": true,   // <--- trailing comma
}
```
Removal of the trailing comma solves the issue.

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

N/A. Let me know if needed and I can supply.

**Comment by iamnbutler** [comment]
This is the correct behavior. It sucks, but it isn't a defect.

We've discussed moving settings to another format, like JSON5, and I think there may be duplicate issues discussing this.

If you want to propose allowing trailing commas in regular JSON, I'd update your title to reflect that.

**Comment by tgross35** [comment]
Quick summary of the JSON versions:

- Standard JSON (`.json`), no comments or trailing commas
- JSONC (`.json` or `.jsonc`), yes comments but no trailing commas
- JSON with comments and trailing commas (no specific extension), I think this is what VS code uses for config (they just call the syntax highlighting "JSON with Comments")
- JSON5 (`.json` or `.json5`), comments, trailing commas, unquoted keys, single quoted strings, number types...

I think it's probably easiest to have config files use JSON5 (keep the `.json` extension) since it's a coherent specification with parsers already available. The "JSON + comments + trailing commas" option really isn't a known format

**Comment by osiewicz** [comment]
This was just fixed in 0.92.3. Give it a go. :)

**Comment by maxbrunsfeld** [comment]
Going to close this out, because Zed now allows trailing commas (as well as C-style comments) in its JSON config files. We're not going to switch to an alternative language like JSON5 for now, because JSON5 is not as well supported by language servers.
